src/VisualStudio/Core/Def/Implementation/LanguageClient/InProcLanguageServer.cs

```diff
@@ -365,14 +365,16 @@ static LanguageServer.Protocol.Diagnostic ConvertToLspDiagnostic(DiagnosticData
 
         private bool IncludeDiagnostic(DiagnosticData diagnostic)
         {
-            if (!diagnostic.Properties.TryGetValue(nameof(DocumentPropertiesService.DiagnosticsLspClientName), out var diagnosticClientName))
-            {
-                // This diagnostic is not restricted to a specific LSP client, so just pass it through
-                return true;
-            }
-
-            // We only include this diagnostic if it directly matches our name.
-            return diagnosticClientName == _clientName;
+            // Some diagnostics only apply to certain clients, e.g. Razor.
+            // If the DocumentPropertiesService.DiagnosticsLspClientName exists, we only include the
+            // diagnostic if it directly matches the client name.
+            // If the DocumentPropertiesService.DiagnosticsLspClientName property doesn't exist,
+            // we know that the diagnostic we're working with is contained in a C#/VB file, since
+            // if we were working with a non-C#/VB file, then the property should have been populated.
+            // In this case, unless we have a null client name, we don't want to publish the diagnostic
+            // (since a null client name represents the C#/VB language client).
+            return DictionaryExtensions.GetOrDefault(
```

Comment:
This should be rewritten back to the extension method form, not sure why it got lost but should be usable.